Create Getting started with List draft (#1925)
DevExpress/devextreme-documentation: concepts/05 UI Components/List/00 Getting Started with List/05 Group Items.md

An alternative to the **key** + **items** structure is a flat array grouped using the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Wrap your array in a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) as shown below and use the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to specify a data field to group data by.

---
##### jQuery

    <!-- tab: index.js -->
    $(function () {
        $('#list').dxList({
            dataSource: new DevExpress.data.DataSource({
                store: products,
                group: 'Category',
            }),
            grouped: true,
        });    
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-list...

>         [grouped]="true">

Reviewer: ```suggestion
    <dx-list...
        [dataSource]="dataSource"
        [grouped]="true">
```

    </dx-list>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';
    import DataSource from "devextreme/data/data_source";

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        // ...

>         dataSource = new DataSource({

Reviewer: In your source code, the `dataSource` is initialized in the constructor. Also, I think we should show the full code to make clear where `products` come from.

            store: this.products,
            group: "Category"
        });
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div id="app-container">

>             <DxList...

Reviewer: ```suggestion
            <DxList...
                :data-source="dataSource"
```

                :grouped="true">
            </DxList>
        </div>
    </template>

    <script>
    // ...

>     import service from './products.service';

Reviewer: Differs from [original](https://github.com/czernous-devexpress/Getting-started-with-List/blob/master/vue/src/App.vue#L18). Let's also show the import of `DataSource`

    export default {
        // ...
        data() {
            // ...
            dataSource: new DataSource({
                store: products,
                group: "Category"

>             }),

Reviewer: ```suggestion
            return {
                dataSource: new DataSource({
                    store: products,
                    group: "Category"
                }),
            }
```

        },
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';

    import 'devextreme/dist/css/dx.common.css';

>     import 'devextreme/dist/css/dx.light.css';

Reviewer: Let's show where `products` and `DataSource` come from.

    const dataSource = new DataSource({
        store: products,
        group: "Category"
    });

    function App() {
        return (
            <div className="App">

>                 <List...

Reviewer: ```suggestion
                <List...
                    dataSource={dataSource}
```

An alternative to the **key** + **items** structure is a flat array grouped using the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Wrap your array in a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) as shown below and use the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to specify a data field to group data by.

---
##### jQuery

    <!-- tab: index.js -->
    $(function () {
        $('#list').dxList({
            dataSource: new DevExpress.data.DataSource({
                store: products,
                group: 'Category',
            }),

>             displayExpr: 'Name',

Reviewer: We already shown `displayExpr`.

```suggestion
```

            grouped: true,
        });    
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-list...

>         displayExpr="Name"

Reviewer: ```suggestion

```

>         grouped="true">

Reviewer: ```suggestion
        [grouped]="true">
```

    </dx-list>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';
    import DataSource from "devextreme/data/data_source";

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        // ...
        dataSource = new DataSource({
            store: this.products,
            group: "Category"
        });
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div id="app-container">
            <DxList...

>                 displayExpr="Name"

Reviewer: ```suggestion

```

                :grouped="true">
            </DxList>
        </div>
    </template>

    <script>
    // ...
    import service from './products.service';

    export default {
        // ...
        data() {
            // ...
            dataSource: new DataSource({
                store: products,
                group: "Category"
            }),
        },
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    const dataSource = new DataSource({
        store: products,
        group: "Category"
    });

    function App() {
        return (
            <div className="App">
                <List...

>                     displayExpr="Name"

Reviewer: ```suggestion
```

> The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a hierarchical array of objects. Each object must contain a **key** property (parent) and an **items** property (child items). Refer to the following help topic for more information: [Grouping in the Data Source](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/).

Reviewer 2: A technical detail issue - the `key` and `items` properties are not necessarily required, so I would write about it in the same way as the referenced topic. These fields facilitate automatic recognition, but don't have to be named like that. 

```suggestion
The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a hierarchical array of objects. The control recognizes a hierarchy if objects contain properties named **key** and nested lists named **items**. Refer to the following help topic for details and for additional ways to build a hierarchy: [Grouping in the Data Source](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/).
```

> The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a hierarchical array of objects. The control recognizes a hierarchy if objects contain properties named **key** and nested lists named **items**. Refer to the following help topic for details and for additional ways to build a hierarchy: [Grouping in the Data Source](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/).

Reviewer 3: ```suggestion
The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a hierarchical array of objects. The control recognizes a hierarchy if objects contain properties named **key** and nested lists named **items**. Refer to the following help topic for more information and additional ways to build a hierarchy: [Grouping in the Data Source](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/).
```

> An alternative to the **key** + **items** structure is a flat array grouped using the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Wrap your array in a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) as shown below and use the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to specify a data field to group data by.

Reviewer 2: `grouped using` - it's clear enough, but does break our 'ing' guidelines and is easy to rewrite. 

You also end the sentence with a preposition. Avoid that if possible. 

And smaller sentences also help. 

 ```suggestion
You can also display a hierarchy in a list bound to a flat array. In this case, use the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Wrap your array in a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) as shown below. Assign the grouping field to the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property.
```

            }),
            grouped: true,
        });    
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-list...
        [dataSource]="dataSource"
        [grouped]="true">
    </dx-list>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';
    import DataSource from "devextreme/data/data_source";
    import { Product, ProductsService } from './products.service';

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        dataSource: DataSource;
        products: Product[] = [];

        constructor(service: ProductsService) {
            this.products = service.getProducts();
            this.dataSource = new DataSource({
                store: this.products,
                group: "Category"
            });
        }
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div id="app-container">
            <DxList...
                :data-source="dataSource"
                :grouped="true">
            </DxList>
        </div>
    </template>

    <script>
    // ...

>     import { prdocuts } from './products';

Reviewer: ```suggestion
    import { products } from './products';
```

> The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true**. The component can automatically group items when the data is structured as a multidimensional [array of objects](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/). Each object must contain a unique **key** value and an array of list **items**. An alternative to the **key** + **items** structure is a flat array which is used in this example. In this case use the **DataSource**'s **store** property as shown in the [Bind the List to Data](/Documentation/Guide/UI_Components/List/Getting_Started_with_List/#Bind_the_List_to_Data) article. Define the [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property within the **DataSource** instance, then specify a field name witihin your data object to group data by.

Reviewer: ```suggestion
The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a [hierarchical array of objects](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/). Each object must contain a **key** property (parent) and an **items** property (child items).

An alternative to the **key** + **items** structure is a flat array grouped using the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Wrap your array in a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) as shown below and use the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to specify a data field to group data by.
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function () {
        $('#list1').dxList({
            dataSource: new DevExpress.data.DataSource({
                // ...
                group: 'Category',
            }),
            displayExpr: 'Name',

>         });

Reviewer: We need to change the code sample according to the suggestions above.

> The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a [hierarchical array of objects](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/). Each object must contain a **key** property (parent) and an **items** property (child items).

Reviewer: It feels like we should point the reader more obviously to the article with more information.

```suggestion
The **List** can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure that data is structured as a hierarchical array of objects. Each object must contain a **key** property (parent) and an **items** property (child items). Refer to the following help topic for more information: [Grouping in the Data Source](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/).
```
